mmengine/optim/optimizer/apex_optimizer_wrapper.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from contextlib import contextmanager
+
+import torch
+import torch.nn as nn
+
+from mmengine.registry import OPTIM_WRAPPERS
+from .optimizer_wrapper import OptimWrapper
+
+try:
+    import apex.amp as apex_amp
+except ImportError:
+    pass
+
+
+@OPTIM_WRAPPERS.register_module()
+class ApexOptimWrapper(OptimWrapper):
+    """A subclass of :class:`OptimWrapper` that supports automatic mixed
+    precision training based on apex.amp.
+
+    ``ApexOptimWrapper`` provides a unified interface with
+    ``OptimWrapper``, so ``ApexOptimWrapper`` can be used in the same way
+    as ``OptimWrapper``.
+
+    Warnings:
+        ``ApexOptimWrapper`` requires
+            [nvidia apex](https://github.com/NVIDIA/apex).
+
+    Args:
+
+        **kwargs: Keyword arguments passed to OptimWrapper.
+
+    Note:
+        If you use ``IterBasedRunner`` and enable gradient accumulation,
+        the original `max_iters` should be multiplied by
+        ``accumulative_counts``.
+    """
+
+    def __init__(self, opt_level='O1', loss_scale='dynamic', **kwargs):
+        super().__init__(**kwargs)
```

Comment:
```suggestion
    def __init__(self, opt_level='O1', loss_scale='dynamic', **kwargs):
    assert apex_amp is not None, \
        'Apex is not installed. Please check https://github.com/NVIDIA/apex#linux.',
        super().__init__(**kwargs)
```
The assertion logic similar to the line in the test code could be added here.